lib/glimesh/channel_lookups.ex

```diff
@@ -195,6 +195,10 @@ defmodule Glimesh.ChannelLookups do
     |> Repo.preload([:category, :user, :streamer])
   end
 
+  def get(id, preloads \\ []) do
```

Comment:
I had two issues with the `get_channel` method:
1. If there is no such channel, `get_by` will return nil which will throw when piped into the `preload` call
2. It always preloads the same associated rows, even when we don't needs them.

My experience with ORMs are shown the first most common performance issue is not preloading enough, the second most common is preloading too much. This approach both allows preloading exactly what you need and  handles missing rows gracefully. I did not replace the existing get_channel method to limit the scope of my changes.

Reply:
Yeah, there's hundreds of cases where we preload data we likely don't even need.

Reply:
Probably ha, but it might not be a big impact, databases are fast!

My approach is just to "leave the code better than you found it" and I choose this as something to dig into and try to find a better pattern that could be used in the future.